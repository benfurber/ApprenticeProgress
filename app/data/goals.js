const goalsData = {
  [1]: {
    actions: ["1", "2"],
    description: "HELLO",
    score: 1,
    tag: "Team thoughtbot",
    title: "I contribute to thoughtbot being a good place to work",
  },
  [2]: {
    actions: ["1", "2"],
    description: "",
    score: 2,
    tag: "Team thoughtbot",
    title: "I iterate over my existing knowledge effectively",
  },
  [3]: {
    actions: ["1", "2"],
    description: "",
    score: 3,
    tag: "Team thoughtbot",
    title: "I’m a great pair programmer",
  },
  [4]: {
    actions: ["1", "2"],
    description: "",
    score: 1,
    tag: "Team thoughtbot",
    title: "I help others become better ",
  },
  [5]: {
    actions: ["1", "2"],
    description: "",
    score: 3,
    tag: "Team thoughtbot",
    title: "I’m passionate about software/products",
  },
  [6]: {
    actions: ["1", "2"],
    description: "",
    score: 4,
    tag: "Team thoughtbot",
    title: "I set my own direction",
  },
};

export { goalsData };
