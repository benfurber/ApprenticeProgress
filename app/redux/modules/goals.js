import gql from "graphql-tag";

import { client } from "utils";

const addGoals = goals => ({
  type: GET,
  payload: goals,
});

const GET = "goals/get";

const query = {
  query: gql`
    {
      goals {
        id
        actions
        description
        evidence
        score
        tag
        title
      }
    }
  `,
};

const fetchGoals = dispatch => {
  return client.query(query).then(result => {
    dispatch(addGoals(result.data.goals));
  });
};


const goalsReducer = (state, action) => {
  switch (action.type) {
    case GET:
      return { ...state, goals: action.payload };
    default:
      return state;
  }
};

export { fetchGoals, goalsReducer };
