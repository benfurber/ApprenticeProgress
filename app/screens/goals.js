import React, { Component } from "react";
import { Container } from "native-base";
import { Goal } from "components";

export class Goals extends Component {
  getGoals(goalsData) {
    return goalsData.goals.map((goal, index) => {
      return this.displayGoal(goal, index);
    });
  }

  displayGoal(goal, index) {
    return <Goal data={ goal } key={ index } />;
  }

  render() {
    return (
      <Container>
        { this.getGoals(goalsData) }
      </Container>
    );
  }
}

const goalsData = {
  goals: [
    {
      "title": "Goal 1",
      "description": "Desc",
    },
    {
      "title": "Goal 2",
      "description": "Desc",
    },
  ],
};
