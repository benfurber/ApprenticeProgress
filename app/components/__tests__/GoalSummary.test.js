// @flow

import React from "react";
import { shallow } from "enzyme";

import { GoalSummary } from "../GoalSummary";
import { navigationMock } from "mocks";

import { goalsData } from "../../data/goals";

describe("goalSummary", () => {
  it("has a valid snapshot", () => {
    const component = shallow(
      <GoalSummary details={goalsData[0]} navigation={navigationMock} />
    );

    expect(component).toMatchSnapshot();
  });

  it("reacts onPress", () => {
    const onPressMock = jest.fn();
    const component = shallow(
      <GoalSummary
        details={goalsData[2]}
        navigation={navigationMock}
        onPress={onPressMock}
      />
    );

    component.simulate("press");

    expect(onPressMock).toHaveBeenCalled();
  });
});
