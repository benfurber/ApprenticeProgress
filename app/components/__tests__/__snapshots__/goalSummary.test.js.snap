// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`goalSummary Has a valid snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GoalSummary
    data={
      Object {
        "description": "Hello world",
        "title": "Testing title",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(CardItem)
          header={true}
        >
          <Styled(H3)>
            Testing title
          </Styled(H3)>
          <Styled(Badge)>
            <Styled(Text) />
          </Styled(Badge)>
        </Styled(CardItem)>,
        <Styled(CardItem)>
          <Styled(Body)>
            <Styled(Text)>
              Hello world
            </Styled(Text)>
          </Styled(Body)>
        </Styled(CardItem)>,
      ],
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(H3)>
              Testing title
            </Styled(H3)>,
            <Styled(Badge)>
              <Styled(Text) />
            </Styled(Badge)>,
          ],
          "header": true,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Testing title",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "Testing title",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Text) />,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(Body)>
            <Styled(Text)>
              Hello world
            </Styled(Text)>
          </Styled(Body)>,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Text)>
              Hello world
            </Styled(Text)>,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Hello world",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "Hello world",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(CardItem)
            header={true}
          >
            <Styled(H3)>
              Testing title
            </Styled(H3)>
            <Styled(Badge)>
              <Styled(Text) />
            </Styled(Badge)>
          </Styled(CardItem)>,
          <Styled(CardItem)>
            <Styled(Body)>
              <Styled(Text)>
                Hello world
              </Styled(Text)>
            </Styled(Body)>
          </Styled(CardItem)>,
        ],
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(H3)>
                Testing title
              </Styled(H3)>,
              <Styled(Badge)>
                <Styled(Text) />
              </Styled(Badge)>,
            ],
            "header": true,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Testing title",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": "Testing title",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Text) />,
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": undefined,
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Body)>
              <Styled(Text)>
                Hello world
              </Styled(Text)>
            </Styled(Body)>,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Text)>
                Hello world
              </Styled(Text)>,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Hello world",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": "Hello world",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
